src/main/java/org/opensearch/ad/ml/ThresholdingResult.java

```diff
@@ -25,13 +31,44 @@
     private final double confidence;
     private final double rcfScore;
     private long totalUpdates;
+
+    // flag indicating if the anomaly is the start of an anomaly or part of a run of
+    // anomalies
     private boolean startOfAnomaly;
+
+    // flag indicating if the time stamp is in elevated score region to be
+    // considered as anomaly
     private boolean inHighScoreRegion;
+
+    /**
+     * position of the anomaly vis a vis the current time (can be -ve) if anomaly is
+     * detected late, which can and should happen sometime; for shingle size 1; this
+     * is always 0
+     */
     private int relativeIndex;
-    private double[] currentTimeAttribution;
-    private double[] oldValues;
+
+    // a flattened version denoting the basic contribution of each input variable
+    private double[] relevantAttribution;
+
+    // oldValues is related to relativeIndex and startOfAnomaly. Read the same
+    // field comment on AnomalyResult.
+    private double[] pastValues;
+
+    // expected values, currently set to maximum 1 expected. In the future, we
+    // might give different expected values with differently likelihood. So
+    // the two-dimensional array allows us to future-proof our applications.
+    // Also, expected values correspond to oldValues if present or current input
+    // point otherwise. If oldValues is present, it will take effort to show this
+    // on UX since we found an anomaly from the past (old values).
```

Comment:
`oldValues` -> `pastValues`

Reply:
fixed

Reply:
I remember @ohltyler had some concern about this part and we decided to change to some json format like 
```
{
 likehood: {feature id: expected value}
}
```

Maybe the change is for another part ?

Reply:
How about add more details in comments, like the expected value is only calculated for anomalous detection intervals, and will generate expected value for each feature if detector has multiple features?

If RCF model detects anomaly for some interval for multiple times, will we only persist the expected value for the first time? For example, shingle size 4, the first shingle is [t1, t2, t3, t4], second shingle is [t2, t3, t4, t5], RCF detect anomaly for t3 in first shingle, then detect anomaly again for t3 in second shingle, will RCF recalculate expected value for t3 in second shingle and will we store that recalculated expected value?

Reply:
> I remember @ohltyler had some concern about this part and we decided to change to some json format like
> 
> ```
> {
>  likehood: {feature id: expected value}
> }
> ```
> 
> Maybe the change is for another part ?

The changes are done in AnomalyResult class.

Reply:
> How about add more details in comments, like the expected value is only calculated for anomalous detection intervals, and will generate expected value for each feature if detector has multiple features?

added details and comments.

> 
> If RCF model detects anomaly for some interval for multiple times, will we only persist the expected value for the first time? For example, shingle size 4, the first shingle is [t1, t2, t3, t4], second shingle is [t2, t3, t4, t5], RCF detect anomaly for t3 in first shingle, then detect anomaly again for t3 in second shingle, will RCF recalculate expected value for t3 in second shingle and will we store that recalculated expected value?

RCF will recalculate expected value for t3 and we will store that.